tests/x11/smt_disconnect_prepare.pm

```diff
@@ -0,0 +1,69 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Do some prepare for internal and external SMT on disconnect SMT case
+# Maintainer: Dehai Kong <dhkong@suse.com>
+
+use base "x11test";
+use strict;
+use testapi;
+use utils;
+use migration;
+use mm_network;
+use mm_tests;
+use suseconnect_register;
+
+sub run {
+    select_console("root-console");
+    if (check_var("SMT", "internal")) {
+        configure_static_network('10.0.2.100/24');
+    }
+    else {
+        configure_static_network('10.0.2.111/24');
+    }
+
+    # server running smt must be registered
+    command_register(get_required_var('VERSION'));
+
+    systemctl("disable SuSEfirewall2");
```

Comment:
I guess we have to keep in mind that SLE15 will get firewalld instead. Let's do something we've done for wicked: (see tests/wicked/wicked_before_test.pm) I guess we can extract this functionality.

Reply:
As this case is only running on SLE12. I think it's fine for this case. I will implement RMT case on SLE15 after released, thanks!

Reply:
Good point, I've me that. Thanks!